modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2432,56 +2435,53 @@ def _apply_func_to_range_partitioning(
                 # simply combine all partitions and apply the sorting to the whole dataframe
                 return self.combine_and_apply(func=func)
 
-        if is_numeric_dtype(self.dtypes[key_column]):
-            method = "linear"
+            if ideal_num_new_partitions < len(self._partitions):
+                if len(self._partitions) % ideal_num_new_partitions == 0:
+                    joining_partitions = np.split(
+                        self._partitions, ideal_num_new_partitions
+                    )
+                else:
+                    joining_partitions = np.split(
+                        self._partitions,
+                        range(
+                            0,
+                            len(self._partitions),
+                            round(len(self._partitions) / ideal_num_new_partitions),
```

Comment:
Well, I didn't write the code, but just moved it a few lines above. My understanding, is that we want for `ideal_num_new_partitions` to be at least 1.5 times less in order to shrink partitioning, as far as I understand, this is done so we wouldn't shrink too much in case the `len(parts)` is just slightly bigger than `ideal_num_parts`:
```python
>>> step = math.ceil(18 / 17)
>>> len(np.split(arr, range(step, 18, step)))
9
>>> step = round(18 / 17)
>>> len(np.split(arr, range(step, 18, step)))
18
```

but just in case, ping @RehanSD as an author of this code